src/extensions/lock_time/mod.rs

```diff
@@ -0,0 +1,172 @@
+            self.update_data(address.as_str(), block, &cellbase, &mut batch)?;
+        }
+
+        self.mature_others(&address, &mut batch)?;
+        batch.commit()?;
+
+        Ok(())
+    }
+
+    fn rollback(&self, tip_number: BlockNumber, tip_hash: &packed::Byte32) -> Result<()> {
+        let block_key = Key::Block(tip_number, tip_hash).into_vec();
+        let raw_data = self
+            .store
+            .get(&block_key)?
+            .expect("ACP extension rollback data is not exist");
+
+        let cellbase_with_address = deserialize::<CellbaseWithAddress>(&raw_data).unwrap();
+        let mut account = self.get_cellbase_account(&cellbase_with_address.address)?;
+        account.remove(&cellbase_with_address.cellbase);
+        account.mature();
+
+        let mut batch = self.get_batch()?;
+        batch.put_kv(
+            Key::CkbAddress(&cellbase_with_address.address).into_vec(),
+            Value::CellbaseCapacity(serialize(&account).unwrap()),
+        )?;
+        batch.commit()?;
+
+        Ok(())
+    }
+
+    fn prune(
+        &self,
+        tip_number: BlockNumber,
+        _tip_hash: &packed::Byte32,
+        keep_num: u64,
+    ) -> Result<()> {
+        let prune_to_block = tip_number - keep_num;
+        let block_key_prefix = vec![KeyPrefix::Block as u8];
+        let mut batch = self.get_batch()?;
+
+        let block_iter = self
+            .store
+            .iter(&block_key_prefix, IteratorDirection::Forward)?
+            .filter(|(key, _v)| {
+                key.starts_with(&block_key_prefix)
+                    && BlockNumber::from_be_bytes(to_fixed_array(&key[1..9])) < prune_to_block
+            });
+
+        for (key, _val) in block_iter {
+            batch.delete(key)?;
+        }
+
+        batch.commit()?;
+
+        Ok(())
+    }
+}
+
+impl<S: Store, BS: Store> LocktimeExtension<S, BS> {
+    pub fn new(
+        store: S,
+        _indexer: Arc<Indexer<BS>>,
+        net_ty: NetworkType,
+        _config: HashMap<String, DeployedScriptConfig>,
+    ) -> Self {
+        LocktimeExtension {
+            store,
+            _indexer,
+            net_ty,
+            _config,
+        }
+    }
+
+    fn get_cellbase_account(&self, addr: &str) -> Result<CellbaseCkbAccount> {
+        if let Some(bytes) = self.store.get(Key::CkbAddress(addr).into_vec())? {
+            let account = deserialize::<CellbaseCkbAccount>(&bytes).unwrap();
+            Ok(account)
+        } else {
+            Ok(Default::default())
+        }
+    }
+
+    fn update_data(
+        &self,
+        addr: &str,
+        block: &BlockView,
+        cellbase: &packed::CellOutput,
+        batch: &mut S::Batch,
+    ) -> Result<()> {
+        let capacity: u64 = cellbase.capacity().unpack();
+        let cellbase_ckb = CellbaseCkb::new(block.epoch().to_rational(), capacity);
+        let mut account = self.get_cellbase_account(addr)?;
+        account.push(cellbase_ckb.clone());
+        account.mature();
```

Comment:
I suggest processing mature altogether, not apart like mature cellbase's account in update_data function and then call a mature_others function